Userland/Libraries/LibAudio/WavLoader.cpp

```diff
@@ -34,11 +35,12 @@ WavLoaderPlugin::WavLoaderPlugin(const StringView& path)
 
 WavLoaderPlugin::WavLoaderPlugin(const ByteBuffer& buffer)
 {
-    m_stream = make<InputMemoryStream>(buffer);
     if (!m_stream) {
         m_error_string = String::formatted("Can't open memory stream");
         return;
     }
+    m_stream = make<InputMemoryStream>(buffer);
+    m_memory_stream = dynamic_cast<InputMemoryStream*>(m_stream.ptr());
```

Comment:
Is this `dynamic_cast` actually necessary?
can this object be anything but an `InputMemoryStream`?

Reply:
Since `m_stream` is of type `OwnPtr<AK::InputStream>` (the parent class of `AK::InputMemoryStream`), the `dynamic_cast` is needed to get a pointer to the child class.

I feel like there is a better way to do this that avoids the `dynamic_cast`, but I'm not smart enough to figure it out :(

Reply:
sure, just `static_cast<InputMemoryStream*>` :P

Reply:
Actually, this might work:

```cpp
    auto memory_stream = make<InputMemoryStream>(buffer);
    m_memory_stream = memory_stream.ptr();
    m_stream = move(memory_stream);
```

Reply:
> sure, just static_cast<InputMemoryStream*> :P

Oh, I guess you're right, that works too.

I'm not sure why I thought I had to `dynamic_cast`.